src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/StaticHolderTypes.cs

```diff
@@ -73,13 +73,18 @@ public override void Initialize(AnalysisContext context)
         private static void AnalyzeSymbol(SymbolAnalysisContext context)
         {
             var symbol = (INamedTypeSymbol)context.Symbol;
+
             if (!symbol.IsStatic &&
```

Comment:
I have reordered the checks to postpone more expensive checks to later in the code. I am not sure if `IsStaticHolderType` is more expensive than `MatchesConfiguredVisibility` or not so feel free to tell me if I shall swap them.